lib/Dredd.js

```diff
@@ -40,215 +99,79 @@ class Dredd {
       end: 0,
       duration: 0,
     };
-    this.files = [];
-    this.transactions = [];
-    this.runner = new Runner(this.configuration);
+    this.transactionRunner = new TransactionRunner(this.configuration);
     this.logger = logger;
-    configureReporters(this.configuration, this.stats, this.tests, this.runner);
-    this.logProxySettings();
   }
 
-  logProxySettings() {
-    const proxySettings = [];
-    for (const envVariableName of Object.keys(process.env || {})) {
-      const envVariableValue = process.env[envVariableName];
-      if (!Array.from(PROXY_ENV_VARIABLES).includes(envVariableName.toUpperCase())) { continue; }
-      if (envVariableValue === '') { continue; }
-      proxySettings.push(`${envVariableName}=${envVariableValue}`);
+  prepareAPIdescriptions(callback) {
+    this.logger.debug('Resolving locations of API description documents');
+    let locations;
+    try {
+      locations = resolveLocations(this.configuration.custom.cwd, this.configuration.options.path);
+    } catch (error) {
+      process.nextTick(() => callback(error));
+      return;
     }
 
-    if (proxySettings.length) {
-      const message = `
-HTTP(S) proxy specified by environment variables: \
-${proxySettings.join(', ')}. Please read documentation on how
-Dredd works with proxies:
-https://dredd.org/en/latest/how-it-works/#using-https-proxy
-`;
-      this.logger.debug(message);
-    }
+    async.waterfall([
+      (next) => {
+        this.logger.debug('Reading API description documents');
+        readLocations(locations, { http: this.configuration.options.http }, next);
+      },
+      (apiDescriptions, next) => {
+        const allAPIdescriptions = this.configuration.apiDescriptions.concat(apiDescriptions);
+        this.logger.debug('Parsing API description documents');
+        parseContent(allAPIdescriptions, next);
+      },
+    ], (error, apiDescriptions) => {
+      if (error) { callback(error); return; }
+
+      this.logger.debug('Compiling HTTP transactions from API description documents');
+      let apiDescriptionsWithTransactions;
+      try {
+        apiDescriptionsWithTransactions = compileTransactions(apiDescriptions);
+      } catch (compileErr) {
+        callback(compileErr);
+        return;
+      }
+
+      callback(null, apiDescriptionsWithTransactions);
+    });
   }
 
   run(callback) {
```

Comment:
The Dredd class represents an object with just a single (asynchronous...) method, run(). That's an antipattern and a clear indication it doesn't need to be a class at all. It is interconnected by references with reporters though and it's Dredd's public JavaScript API, so I'm leaving it as a class for now. Also for now ignoring the fact that it calls the callback with both error and the stats (fixing it would be a breaking change). In the future I can imagine something like `const stats = await dredd(config);`